backend/bcssm_backend/user_queries.py

```
@@ -0,0 +1,67 @@
+    rows = _db.execute_readonly_query(query)
+    return [row[0] for row in rows]
+
+
+def _fetch_user_info(where_clause, params, log_identifier):
+    # where_clause must be a hard-coded SQL fragment (e.g. "u.name = :user_name");
+    # all user input goes through params so execute_readonly_query keeps it parameterized.
+    query = (
+        "SELECT u.name, u.role, s.name AS section_name "
+        "FROM users u "
+        "LEFT JOIN sections s ON u.section_id = s.id "
+        f"WHERE {where_clause};"
```

Ruff (0.15.17)

[error] Possible SQL injection vector through string-based query construction

(S608)